overlord/snapstate/handlers_link_test.go

```diff
@@ -556,6 +556,7 @@ func (s *linkSnapSuite) TestDoLinkSnapSuccessSnapdRestartsOnClassic(c *C) {
 	t := s.state.NewTask("link-snap", "test")
 	t.Set("snap-setup", &snapstate.SnapSetup{
 		SideInfo: si,
+		Type:     snap.TypeSnapd,
```

Comment:
We seem to be inconsistent with the use of snap.Type* vs string in these tests (it seems that the "as string" approach prevails and I vaguely remember was preferred; fwtw I only see 3 instances of Type: snap.Type* assignments in these tests.
That's mostly an observation to keep in mind and possibly address in a future PR.

Reply:
SnapSetup.Type  is a snap.Type while SnapState.SnapType is a string, iirc

Reply:
Right. Makes sense to keep it like this. We're inconsistent in other places then.